src/dirac_cwl_proto/execution_hooks/plugins/core.py

```diff
@@ -39,12 +45,95 @@ class QueryBasedPlugin(ExecutionHooksBasePlugin):
+            return Path(self.query_root) / Path(*path_parts) / Path(input_name)
+
+        return Path(self.query_root) / Path(input_name)
+
+    def get_output_query(self, output_name: str, **kwargs: Any) -> Optional[Path]:
+        """Generate LFN-based output path.
+
+        Accepts and ignores extra kwargs for interface compatibility.
+        """
+        # Output path: /grid/data/vo/outputs/campaign/site
+        output_base = Path(self.query_root)
+        if self.vo:
+            output_base = output_base / self.vo
+        output_base = output_base / "outputs"
+
+        if self.campaign:
+            output_base = output_base / self.campaign
+        if self.site:
+            output_base = output_base / self.site
+
+        return output_base
+
+    def store_output(
+        self,
+        output_name: str,
+        src_path: str | Path | Sequence[str | Path],
+        **kwargs: Any,
+    ) -> None:
+        """Store an output file or set of files via the appropriate storage interface.
+
+        :param str output_name:
+            The logical name of the output to store, used to determine the storage
+            target (sandbox or output path).
+        :param str | Path | Sequence[str | Path] src_path:
+            The path or list of paths to the source file(s) to be stored.
+            Can be a single path (string or Path) or a sequence of paths.
+        :param Any **kwargs:
+            Additional keyword arguments for extensibility.
+        """
+        logger.info(f"Storing output {output_name}, with source {src_path}")
+
+        if not src_path:
+            raise RuntimeError(
+                f"src_path parameter required for filesystem storage of {output_name}"
+            )
+        if self.output_sandbox and output_name in self.output_sandbox:
+            if isinstance(src_path, Path) or isinstance(src_path, str):
+                src_path = [src_path]
+            self._sandbox_store_client.uploadFilesAsSandbox(src_path)
+        else:
+            if self.output_paths and output_name in self.output_paths:
+                lfn = self.output_paths[output_name]
+            else:
+                lfn = self.get_output_query(output_name)
+
+            if lfn:
+                if isinstance(src_path, str) or isinstance(src_path, Path):
+                    src_path = [src_path]
+                for src in src_path:
+                    file_lfn = Path(lfn) / Path(src).name
+                    res = None
+                    for se in self.output_se:
+                        res = returnSingleResult(
+                            self._datamanager.putAndRegister(str(file_lfn), src, se)
+                        )
+                        if res["OK"]:
+                            break
+                    if res and not res["OK"]:
+                        raise RuntimeError(
+                            f"Could not save file {src} with LFN {str(lfn)} : {res['Message']}"
```

Comment:
Do we want to specify `store_output()` in the `QueryBasedPlugin`?
I haven't checked carefully (so please let me know if I'm wrong), but it seems to be a copy paste of what is defined in `ExecutionHooksHint`.

If it's a duplicate, I would remove it from here.